src/components/AudioPlayer.tsx

```diff
@@ -86,6 +92,8 @@ export default function AudioPlayer({ durationGuess, src, title, trackType }: Au
 			audioRef.current.pause();
 		} else {
 			audioRef.current.play().catch();
+			prev?.current.pause();
```

Comment:
restarting same song breaks unless u have this check or something similar but it works otherwise basicallyu
```ts
			if (prev?.current && prev.current !== audioRef.current)
				prev.current.pause();
```